query/node.go

```diff
@@ -88,9 +88,10 @@ type functionExpression struct {
 // ---------------
 // These nodes are only present during the parsing step and are not present
 // in the resulting command.
-// There are two types of temporary nodes:
+// There are three types of temporary nodes:
 // * literals (constants in the syntax tree).
 // * lists
+// * evaluation context nodes
```

Comment:
Although they initially were intended to be, they aren't actually temporary anymore.